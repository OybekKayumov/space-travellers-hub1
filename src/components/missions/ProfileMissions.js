import { useDispatch } from 'react-redux';
import { PropTypes } from 'prop-types';
import Button from 'react-bootstrap/Button';
import { leaveMission } from '../../redux/missions/missions';

const ProfileMissions = (props) => {
  const dispatch = useDispatch();
  const { id, name } = props;

  const handleLeaveMission = () => {
    dispatch(leaveMission(id));
  };

  return (
    <>
      <th>{name}</th>
      <td><Button type="button" variant="danger" className="remove-reservation" onClick={handleLeaveMission}>Leave Mission</Button></td>
    </>
  );
};

ProfileMissions.propTypes = {
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};

export default ProfileMissions;
