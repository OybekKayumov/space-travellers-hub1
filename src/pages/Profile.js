import React from 'react';
import { useSelector } from 'react-redux';
import 'bootstrap/dist/css/bootstrap.min.css';
import Table from 'react-bootstrap/Table';
import ProfileRockets from '../components/rockets/ProfileRockets';
import ProfileMissions from '../components/missions/ProfileMissions';
import style from './Profile.module.css';

export default function Profile() {
  const missionList = useSelector((state) => state.missions);
  const profileMission = missionList.filter((mission) => mission.reserved === true);

  const rocketList = useSelector((state) => state.rockets);
  const profileRocket = rocketList.filter((rocket) => rocket.reservation === true);

  return (
    <div className={style.flex}>
      <Table bordered hover>
        <thead>
          <tr>
            <th
              style={{
                padding: '5px 10px', backgroundColor: 'white',
              }}
              colSpan={3}
            >
              My Rocket Reservations
            </th>
          </tr>
        </thead>
        <tbody>
          {profileRocket.length ? (
            profileRocket.map((item) => (
              <ProfileRockets
                key={item.id}
                id={item.id}
                name={item.name}
                wikipedia={item.wikipedia}
              />
            ))
          ) : (
            <tr className="empty"><td>No Rockets have been reserved</td></tr>
          )}
        </tbody>
      </Table>
      <Table bordered hover>
        <thead>
          <tr>
            <th colSpan={2}>My Missions</th>
          </tr>
        </thead>
        <tbody>
          {profileMission.length ? (
            <tr>
              {profileMission.map((item) => (
                <ProfileMissions
                  key={item.mission_id}
                  id={item.mission_id}
                  name={item.mission_name}
                  description={item.mission_description}
                />
              ))}
            </tr>
          ) : (
            <tr className="empty"><td>You have not joined any missions at the moment</td></tr>
          )}
        </tbody>
      </Table>
    </div>
  );
}
