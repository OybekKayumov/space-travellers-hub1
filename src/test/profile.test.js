import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import elementStore from '../redux/configureStore';
import ProfileRocket from '../components/Rockets/ProfileRockets';
import ProfileMissions from '../components/missions/ProfileMissions';
import Profile from '../pages/Profile';

const rocket = [
  {
    key: 'rocket',
    id: 'rocket',
    name: 'rocket 1',
    wikipedia: 'http://en.wikipedia.org/',
  },
];

const mission = [
  {
    key: 'mission',
    id: 'mission',
    name: 'mission 1',
    description: 'mission alpha',
  },
];

describe('test render the profile component', () => {
  it('profile section renders a hardcoded obj, passed as a prop', () => {
    // act
    const { container } = render(
      <Provider store={elementStore}>
        <Profile>
          <ProfileRocket key={rocket[0].id} rocket={rocket[0]} />
          <ProfileMissions key={mission[0].id} mission={mission[0]} />
        </Profile>
      </Provider>,
    );

    // assert
    expect(container).toMatchSnapshot();
  });
});
