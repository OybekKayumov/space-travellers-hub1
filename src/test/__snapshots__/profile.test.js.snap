// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render the profile component profile section renders a hardcoded obj, passed as a prop 1`] = `
<div>
  <div
    class="flex"
  >
    <table
      class="table table-bordered table-hover"
    >
      <thead>
        <tr>
          <th
            colspan="3"
            style="padding: 5px 10px; background-color: white;"
          >
            My Rocket Reservations
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="empty"
        >
          <td>
            No Rockets have been reserved
          </td>
        </tr>
      </tbody>
    </table>
    <table
      class="table table-bordered table-hover"
    >
      <thead>
        <tr>
          <th
            colspan="2"
          >
            My Missions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="empty"
        >
          <td>
            You have not joined any missions at the moment
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
